lib/Models/InitSource.ts

```diff
@@ -20,12 +21,12 @@ interface InitOptions {
   options: InitSource[];
 }
 
-type InitSource = { name?: string } & (
-  | InitUrl
-  | InitData
-  | InitOptions
-  | InitDataPromise
-);
+type InitSource = {
+  /** Name is only used for debugging purposes */
+  name?: string;
+  /** Severity to use for errors caught while loading/applying this initSource */
+  errorSeverity?: TerriaErrorSeverity;
+} & (InitUrl | InitData | InitOptions | InitDataPromise);
```

Comment:
`TerriaErrorSeverity` is needed here so we can different severity for different init sources - for example - shareData is `Warning`